src/bun.js/bindings/bindings.zig

```diff
@@ -4183,6 +4191,27 @@ pub const JSValue = enum(JSValueReprInt) {
         return jsType(this).isStringLike();
     }
 
+    /// Returns true only for string literals
+    /// - `" string literal"`
+    pub inline fn isStringLiteral(this: JSValue) bool {
+        if (!this.isCell()) {
+            return false;
+        }
+
+        return jsType(this).isString();
+    }
+
+    /// Returns true if
+    /// - `new String("123")`
+    /// - `class DerivedString extends String; new DerivedString("123")`
+    pub inline fn isStringObjectLike(this: JSValue) bool {
+        if (!this.isCell()) {
+            return false;
+        }
+
+        return jsType(this).isStringObject() or jsType(this).isDerivedStringObject();
```

Comment:
This should be in `JSType` and then this should call `this.jsType().isStringObjectLike()`

Reply:
because this calls `jsType(this)` twice when it definitely doesn't need to

Reply:
updated